--- bin/idea.sh.orig	2012-12-07 17:44:30.987002000 +0700
+++ bin/idea.sh	2012-12-07 18:55:53.886991000 +0700
@@ -20,6 +20,8 @@
   exit 1
 fi
 
+. /etc/conf.d/idea-12
+
 OS_TYPE=`"$UNAME" -s`
 
 # ---------------------------------------------------------------------
@@ -97,7 +99,7 @@
 # ---------------------------------------------------------------------
 # COMMENT LINE BELOW TO REMOVE PAUSE AFTER OPEN JDK WARNING
 # ---------------------------------------------------------------------
-  read IGNORE
+#  read IGNORE
 fi
 if [ $BITS -eq 0 ]; then
   BITS="64"
@@ -149,7 +151,7 @@
   fi
 fi
 
-COMMON_JVM_ARGS="\"-Xbootclasspath/a:$IDE_HOME/lib/boot.jar\" -Didea.paths.selector=IntelliJIdea12 $IDE_PROPERTIES_PROPERTY"
+COMMON_JVM_ARGS="\"-Xbootclasspath/a:$IDE_HOME/lib/boot.jar\" -Didea.home.path=\"${IDE_HOME}\" -Didea.paths.selector=IntelliJIdea12 $IDE_PROPERTIES_PROPERTY"
 IDE_JVM_ARGS=""
 ALL_JVM_ARGS="$VM_OPTIONS $COMMON_JVM_ARGS $IDE_JVM_ARGS $AGENT $REQUIRED_JVM_ARGS"
 
