--- bin/idea.sh.origin	2012-07-27 15:07:55.881429424 +0700
+++ bin/idea.sh	2012-07-27 15:08:09.510428517 +0700
@@ -82,7 +82,7 @@
 # ---------------------------------------------------------------------
 # COMMENT LINE BELOW TO REMOVE PAUSE AFTER OPEN JDK WARNING
 # ---------------------------------------------------------------------
-  read IGNORE
+#  read IGNORE
 fi
 if [ $BITS -eq 0 ]; then
   BITS="64"
